src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -484,7 +484,7 @@ export class ShellIntegrationAddon extends Disposable implements IShellIntegrati
 				return true;
 			}
 			case VSCodeOscPt.EnvSingleStart: {
-				this._createOrGetShellEnvDetection().startEnvironmentSingleVar(args[0] === this._nonce);
+				this._createOrGetShellEnvDetection().startEnvironmentSingleVar(args[0] === '1', args[1] === this._nonce);
```

Comment:
Clear is optional now, so the nonce could be in args[0] or args[1] depending on args.length

Reply:
I thought args.length always be 1 if we made clear optional and not pass in 0 for bash, zsh?
- Length 1 for bash,zsh for its nonce
- Length 1 for fish for its clear argument (there is no nonce in fish script so I dont even bother passing this in)